crates/module-system/module-implementations/sov-evm/src/call.rs

```diff
@@ -46,84 +48,36 @@ where
         // Inside the EVM, we use nonces only for the CREATE operation.
         // The uniqueness check was performed before the call was dispatched.
         let account_nonce = self.get_account_nonce(signer, state)?;
-
-        let evm_tx: TransactionSigned = convert_to_transaction_signed(message.rlp)?;
-
+        let gas_limit = state.gas_limit()?;
```

Comment:
The sov gas meter is initialized with:

1. `remaining_gas: S::Gas`
2. `remaining_funds: Option<Amount>`

Initially, `remaining_funds` is set to `tx.max_fee`. Since Ethereum also uses a `max_fee` concept, we should be able to support this as well. Maybe it works out of the box and we just need to replace the `tx.max_fee` with the one from `gas_meter.`

Can be done as a separate PR.